### PR TITLE
Nested folders: Fix move

### PR DESCRIPTION
**Pull request by papagian**
**What is this feature?**

- Fixes `POST /folders/{folder_uid}/move` response to align with the `GET /folders/{folder_uid}` response (JSON tags used to have different casing)

<details>
<summary>Before this change:</summary>

```
~/go/src/github.com/grafana/grafana: curl -H "Content-Type: application/json" -XPOST http: //admin:admin@localhost:3000/api/folders/dd2b7703-0cdc-4537-9593-e6ba0b59ba56/move -d '{"parentUid": "d8bed27b-5e3b-4073-93b1-96c270b5b73b"}' | jq
{
	"ID": 4,
	"OrgID": 1,
	"UID": "dd2b7703-0cdc-4537-9593-e6ba0b59ba56",
	"ParentUID": "d8bed27b-5e3b-4073-93b1-96c270b5b73b",
	"Title": "subfolder",
	"Description": "",
	"Created": "2023-03-28T21:16:58.396459+03:00",
	"Updated": "2023-03-29T16:22:44.557455+03:00",
	"Version": 0,
	"URL": "/dashboards/f/dd2b7703-0cdc-4537-9593-e6ba0b59ba56/subfolder",
	"UpdatedBy": 0,
	"CreatedBy": 0,
	"HasACL": false
}
```
</details/>

<details>
<summary>After this change:</summary>

```
~/go/src/github.com/grafana/grafana: curl -H "Content-Type: application/json" -XPOST http: //admin:admin@localhost:3000/api/folders/dd2b7703-0cdc-4537-9593-e6ba0b59ba56/move -d '{"parentUid": "d8bed27b-5e3b-4073-93b1-96c270b5b73b"}' | jq
{
	"id": 4,
	"uid": "dd2b7703-0cdc-4537-9593-e6ba0b59ba56",
	"title": "subfolder",
	"url": "/dashboards/f/dd2b7703-0cdc-4537-9593-e6ba0b59ba56/subfolder",
	"hasAcl": false,
	"canSave": true,
	"canEdit": true,
	"canAdmin": true,
	"canDelete": true,
	"createdBy": "Anonymous",
	"created": "2023-03-28T21:16:58.396459+03:00",
	"updatedBy": "Anonymous",
	"updated": "2023-03-29T16:23:52.427721+03:00",
	"parentUid": "d8bed27b-5e3b-4073-93b1-96c270b5b73b"
}
```
</details/>

- Allows to move a folder to the root folder
- Adds swagger annotations and re-generates the OpenAPI specifications (note that, it seems that the specifications contain additional changes for other modifications introduced in the API outside the scope of this but the specifications weren't updated as well)

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.
- [ ] There are no known compatibility issues with older supported versions of Grafana, or plugins.
- [ ] It passes the [Hosted Grafana feature readiness review](https://docs.google.com/document/d/1QL9Ly8KnXzpb6ISbg49pTODRO5mhA5tkkfIZVX6pqQU/edit) for observability, scalability, performance, and security.

### CODE REVIEW

File: pkg/services/folder/folderimpl/sqlstore_test.go [hunk 1]
```diff
@@ -390,9 +390,10 @@ func TestIntegrationGetParents(t *testing.T) {
 		require.NoError(t, err)
 	})
 
-	t.Run("get parents of unknown folder should return an error", func(t *testing.T) {
-		_, err := folderStore.GetParents(context.Background(), folder.GetParentsQuery{})
-		require.ErrorIs(t, err, folder.ErrFolderNotFound)
+	t.Run("get parents of root folder should should be empty", func(t *testing.T) {
```
Comment (IevaVasiljeva) [inline pkg/services/folder/folderimpl/sqlstore_test.go:393]: nit:
```suggestion
	t.Run("get parents of root folder should be empty", func(t *testing.T) {
```

File: pkg/api/folder_test.go [hunk 2]
```diff
@@ -378,6 +380,14 @@ func TestFolderMoveAPIEndpoint(t *testing.T) {
...
 			},
 		},
+		{
+			description:  "can move folder to the root folder with specific permissions",
+			newParentUid: "",
+			expectedCode: http.StatusOK,
+			permissions: []accesscontrol.Permission{
+				{Action: dashboards.ActionFoldersWrite, Scope: dashboards.ScopeFoldersProvider.GetResourceScopeUID("uid")},
+			},
+		},
```
Comment (IevaVasiljeva) [inline pkg/api/folder_test.go:390]: We actually also require the permission to create a folder to be able to move folder to the root level. This check is done in the folder service: https://github.com/grafana/grafana/blob/main/pkg/services/folder/folderimpl/folder.go#L494-L509

So I think this is not the right place to test for this. In general, it's not great that access control is scattered in several places, and it's something that we want to clean up in the future.

**Review by IevaVasiljeva — APPROVED** [review]
LGTM with a small comment on the tests.

File: pkg/api/folder_test.go (hunk 2 above)
Comment (papagian) [reply pkg/api/folder_test.go:390]: ah I see: we use a fake service there

**Pull request merged**